packages/note/index.d.ts

```diff
@@ -0,0 +1,303 @@
+type Octave = number;
+
+type NoteProps = {
+  name: Note;
+  letter: NoteLetter;
+  acc: NoteAccidental; // {String}: the note accidentals
+  octave: Octave; // {Number}: the octave or null if not present
+  step: number; // {Number}: number equivalent of the note letter. 0 means C ... 6 means B.
+  pc: Note; //{String}: the pitch class (letter + accidentals)
+  alt: number; // {Number}: number equivalent of accidentals (negative are flats, positive sharps)
+  chroma: number; // {Number}: number equivalent of the pitch class, where 0 is C, 1 is C# or Db, 2 is D...
+  midi: Midi; // {Number}: the note midi number (IMPORTANT! it can be outside 0 to 127 range)
+  freq: number; // {Number}: the frequency using an equal temperament at 440Hz
+};
+/**
+ * [![npm version](https://img.shields.io/npm/v/tonal-note.svg)](https://www.npmjs.com/package/tonal-note)
+ * [![tonal](https://img.shields.io/badge/tonal-note-yellow.svg)](https://www.npmjs.com/browse/keyword/tonal)
+ *
+ * `tonal-note` is a collection of functions to manipulate musical notes in scientific notation
+ *
+ * This is part of [tonal](https://www.npmjs.com/package/tonal) music theory library.
+ *
+ * ## Usage
+ *
+ * ```js
+ * import * as Note from "tonal-note"
+ * // or const Note = require("tonal-note")
+ * Note.name("bb2") // => "Bb2"
+ * Note.chroma("bb2") // => 10
+ * Note.midi("a4") // => 69
+ * Note.freq("a4") // => 440
+ * Note.oct("G3") // => 3
+ *
+ * // part of tonal
+ * const Tonal = require("tonal")
+ * // or import Note from "tonal"
+ * Tonal.Note.midi("d4") // => 62
+ * ```
+ *
+ * ## Install
+ *
+ * [![npm install tonal-note](https://nodei.co/npm/tonal-note.png?mini=true)](https://npmjs.org/package/tonal-note/)
+ *
+ * ## API Documentation
+ *
+ * @module Note
+ */
+
+/**
+ * Get a list of note names (pitch classes) within a octave
+ *
+ * @param {string} accTypes - (Optional, by default " b#"). A string with the
+ * accidentals types: " " means no accidental, "#" means sharps, "b" mean flats,
+ * can be combined (see examples)
+ * @return {Array}
+ * @example
+ * Note.names(" b") // => [ "C", "Db", "D", "Eb", "E", "F", "Gb", "G", "Ab", "A", "Bb", "B" ]
+ * Note.names(" #") // => [ "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B" ]
+ */
+export declare const names: (
+  accTypes: " " | "#" | "b" | " #" | " b" | "b#" | "#b"
+) => string[];
+/**
+ * Split a string into tokens related to note parts.
+ * It returns an array of strings `[letter, accidental, octave, modifier]`
+ *
+ * It always returns an array
+ *
+ * @param {String} str
+ * @return {Array} an array of note tokens
+ * @example
+ * Note.tokenize("C#2") // => ["C", "#", "2", ""]
+ * Note.tokenize("Db3 major") // => ["D", "b", "3", "major"]
+ * Note.tokenize("major") // => ["", "", "", "major"]
+ * Note.tokenize("##") // => ["", "##", "", ""]
+ * Note.tokenize() // => ["", "", "", ""]
+ */
+export declare function tokenize(
+  str?: string
+): [string, string, string, string];
```

Comment:
Great! Now with type aliases we can set the first parameter to `NoteLetter`
```suggestion
): [NoteLetter, string, string, string];
```

It would be nice to have a type to represent zero, one or more accidentals... is that possible?

Reply:
Looked into that, can't think of a way to do it right now. I'll think about it more and update if/when I find a way.